core-bundle/src/EventListener/SearchIndexListener.php

```diff
@@ -42,24 +42,30 @@ public function onKernelTerminate(TerminateEvent $event): void
     {
         $request = $event->getRequest();
 
-        // Only index GET requests (see #1194)
+        // Only handle GET requests (see #1194)
         if (!$request->isMethod(Request::METHOD_GET)) {
             return;
         }
 
-        // Do not index fragments
+        // Do not handle fragments
         if (preg_match('~(?:^|/)'.preg_quote($this->fragmentPath, '~').'/~', $request->getPathInfo())) {
             return;
         }
 
         $document = Document::createFromRequestResponse($request, $event->getResponse());
-        $lds = $document->extractJsonLdScripts();
 
-        // If there are no json ld scripts at all, nothing will be indexed
-        if (0 === \count($lds)) {
-            return;
-        }
+        // Index if successful, clear if not
+        if ($event->getResponse()->isSuccessful()) {
+            // If there are no json ld scripts at all, nothing will be indexed
+            $lds = $document->extractJsonLdScripts();
 
-        $this->indexer->index($document);
+            if (0 === \count($lds)) {
+                return;
```

Comment:
Shouldn’t we also delete the document from the index in this case?

Reply:
I thought so too but that would result in useless search index delete commands for 99% of all the use cases (everything that's not coming from Contao in your own Symfony app). The 1% case would be where a page was once generated by Contao and it's now not anymore. I think this is very rare and sounds like you've fundamentally changed your app = you should rebuild the whole search index anyway.

Reply:
If search index delete commands for non-existent entries don’t hurt the performance too much, I think we should delete it in this case too to make it more robust.

Reply:
I really don't know about this. I still think we should keep it as is.

Reply:
I am also in favor of keeping the `return` statement. If there is no JSON data, the request is not a Contao request and the listener should not handle it.

Reply:
> don’t hurt the performance too much

Think of an external search indexer like ES or Algolia. It's an API request for every response. Sure, it's a `kernel.terminate` listener but we still have users that do not use fpm or litespeed.
I will also submit another PR to disable this listener completely (makes no sense to run it if you clean the search index regularly using a cronjob imho).

Reply:
The only unlogical thing about this is that we are not indexing the document if there is no JSON data but we are deleting it no matter whether there is JSON data or not. Is this intended?

Reply:
Right, fixed in a2eb4d762a18f630ccb2d4f2d7469d32f7dc754f.